libraries/eosiolib/contracts/eosio/eosio.hpp

```diff
@@ -9,6 +9,7 @@
 #include "multi_index.hpp"
 #include "dispatcher.hpp"
 #include "contract.hpp"
+#include "../../core/eosio/reflect.hpp"
```

Comment:
Do we have a guideline for what does and does not go into `eosio/eosio.hpp`

This seems like something more appropriately pushed down into the headers (like map.hpp or producer_schedule.hpp). It seems that at least in this header map.hpp precedes the inclusion of reflect.hpp and map.hpp uses the CDT_REFLECT macro.

Reply:
We do.  It is something that would be used by people in a lot of cases.  Historically, eosio.hpp is supposed to be a catchall header for a one header-include for the majority of normal operations.  After this release, the old EOSLIB_SERIALIZE macro is just going to be replaced full sail with the CDT_REFLECT.  They are fundamentally different in how they work, but allow for the same things to be done, but CDT_REFLECT allows for a lot more to be done on custom data types.  Users might have to use it with their custom structs for constructing a key, it is determined by a few separate factors (i.e. PFR fails because it is not a POD type and those types of situations.